### PR TITLE
Add yarn parser

### PR DESCRIPTION
**Pull request by eldemcan**
This change adds fully compatible yarn v1 parser. It only parses yarn.lock file at the moment it doesn't fetch license data

### CODE REVIEW

**Comment by eldemcan** [comment]
@mokhan can you take a look at this one please ? I don't know why some of the tests are failing

File: lib/spandx/js/parsers/yarn.rb [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+
+          lines.each do |line|
+            return true if line.match?(version_regex)
+          end
+
+          false
+        end
+
+        def convert_objects(dependencies)
+          deps = Set.new
```
Comment (xlgmokha) [inline lib/spandx/js/parsers/yarn.rb:227]: Brilliant.

File: lib/spandx/js/parsers/yarn.rb [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+          dependencies.each do |dependency, details|
+            name = dependency.match(/(^.*)(@)(.*)/i).captures.first
```
Comment (xlgmokha) [inline lib/spandx/js/parsers/yarn.rb:229]: An alternative is to use `scan`. Also, named capture groups might make this a little more readable. E.g. (?name)

**Review by xlgmokha — APPROVED** [review]
:shipit:

**Pull request merged**